### PR TITLE
[Android] Do not require certificate transparency

### PR DESCRIPTION
**Pull request by lincsoon**
Certificate transparency is a unique feature of Chrome but non-standard
SSL protocol. It aims to improve the security aloneside with CA. But
Crosswalk is based on content shell so the network connections initiated
by it do not support this certificate policy. This causes a SSL error
called ERT_STATUS_CERTIFICATE_TRANSPARENCY_REQUIRED and makes the app
continue to pop up the "SSL Certificate Error Alert" dialog.

This patch declares that Crosswalk doesn't require certificate
transparency to bypass the SSL error above.

BUG=XWALK-7398

### CODE REVIEW

**Comment by fujunwei** [comment]
Did chromium upstream fixed this issue?
https://codereview.chromium.org/2495583002

**Comment by xzhan96** [comment]
lgtm

**Comment by lincsoon** [comment]
@fujunwei No, the error code in Crosswalk is not same as this issue.

**Comment by fujunwei** [comment]
LGTM

**Comment by rakuco** [comment]
I'd like a more detailed explanation of how Crosswalk and content shell differ from Chromium in this respect. I tried taking a look at this today and couldn't see what sort of "policies" Chromium implements that we do not.

https://m.baidu.com, which was the original URL causing trouble, has a certificate from Symantec, the CA that started the whole CT work upstream. The public key hashes that `net::TransportSecurityState::ShouldRequireCT()` iterates over are different when using Crosswalk/content shell and Chromium (even on Linux), but it is not clear to me why.

In other words: without a bigger explanation, this patch seems to be disabling legitimate checks and fixing the JIRA issue as a side-effect.

**Comment by rakuco** [comment]
> I'd like a more detailed explanation of how Crosswalk and content shell differ from Chromium in this respect. I tried taking a look at this today and couldn't see what sort of "policies" Chromium implements that we do not.

I've also posted this question upstream: https://groups.google.com/a/chromium.org/d/msg/chromium-dev/_ZILH704AVU/ZD0XqL3JCgAJ

**Comment by rakuco** [comment]
... and trying to get additional clarification here: https://bugs.chromium.org/p/chromium/issues/detail?id=669978#c9

**Comment by rakuco** [comment]
Superseded by #3964.
